Ability to toggle ranges via legend

milesap opened the pull request:
## Description
Allows symbology to be toggled. New controls added when symbology is expanded. Symbology is filtered both on the map and table (if open).

Can be disabled in config under layer config as `toggleSymbology: false`. It's true by default.

Do not merge until James gives the :white_check_mark: since geoApi is being beefed up to handle more edge cases on the `definitionClause`.

@dan-bowerman got you on this as well to make sure this works for CESI.

Closes #2368

## Testing
46 `Dynamic Layer with only one child with filter -> Power Plants".  You can also test sample 1 crops but dt is very slow.

## Documentation
inline

## Checklist
<!-- Quick checklist for items that are easy to miss -->

- [x] Commit messages follow [the guidelines](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/master/CONTRIBUTING.md#-git-commit-guidelines)
- [x] Release notes have been updated
- [x] PR targets the correct release version
- [x] Help files and documentation have been updated

Remember, it is a *muffin offence* to open a PR with any of the above checklist items incomplete.

Please keep the original issue up to date with the final solution, expected behaviour, and any additional notes for testers

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/fgpv-vpgf/fgpv-vpgf/2418)
<!-- Reviewable:end -->

james-rae commented:
Some test services for various cases (still trying to find a default symbol renderer one)

1-field unique (warning, service acting flakey, maybe use another)
http://maps-cartes.ec.gc.ca/arcgis/rest/services/CESI/CESI_AirEmissions_CO/MapServer/0

class break
http://section917.cloudapp.net/arcgis/rest/services/TestData/ClassBreaks/MapServer/0

two field unique (with renderer -> legend aggregation)
http://geoappext.nrcan.gc.ca/arcgis/rest/services/NEB/Incident/MapServer/0

tile (no filters)
http://maps-cartes.ec.gc.ca/arcgis/rest/services/Basemaps/OilSandsBasemap/MapServer

simple (no filters)
http://section917.cloudapp.net/arcgis/rest/services/TestData/BasicCity/MapServer/0

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-Kvmqur1GscTRizDbKqB:b9tz1rl)*
<!-- Sent from Reviewable.io -->

dan-bowerman commented:
The English rebuilt CESI services are all ready to roll as well: 
http://section917.cloudapp.net/arcgis/rest/services/CESI

So far, so good on my end! Looks and functions great!

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-KvmrWhbUj6MvtXdaEu_:b-ex2uvy)*
<!-- Sent from Reviewable.io -->

JahidAhmed commented:
All the datatables are empty, is there any reason for that?

---

Review status: 0 of 9 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-Kvms9h1uN3GeO7zBqBy:b9gn1n8)*
<!-- Sent from Reviewable.io -->

dan-bowerman commented:
WOMM? Only the toggled on data is displayed. This sample only works with the Power Plants service. 
![✂-1](https://files.reviewableusercontent.io/images/reviewable/5294596/k314x9TWBS41YVIBRdknVGbr.png)

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-KvmtKzp5NDuq_N53anj:b-fe1avs)*
<!-- Sent from Reviewable.io -->

JahidAhmed commented:
Oh yep, that's working as expected but I wasn't too sure if the other datatables were empty by design somehow? (Don't know too much about this PR)

---

Review status: 0 of 9 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-KvmtnOswP3zRoq3RmTl:b-7tqc1y)*
<!-- Sent from Reviewable.io -->

milesap commented:
I'll fix, some are empty but shouldn't be

---

Review status: 0 of 9 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-Kvmu361GoVYsZf7jOEX:b3h36vt)*
<!-- Sent from Reviewable.io -->

JahidAhmed commented:
Also, when you toggle the entire layer to be off, should the symbology also be toggled off or is it supposed to stay on?

---

Review status: 0 of 9 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-KvmuahYlgUAkL6t-5Mk:b5is4tk)*
<!-- Sent from Reviewable.io -->

milesap commented:
My vote is for the symbology to remain as is rather then reset them but that can be up for debate. I'm wondering if their initial visibility should follow initial layer visibility as it does now or always be on/off.

---

Review status: 0 of 9 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-KvmvPWhNwumK_oJ1ryN:b-jfvo0f)*
<!-- Sent from Reviewable.io -->

james-rae commented:
If we can have a grid open on an invisible layer, then I think we should not tie the checkboxes to layer visibility.  Else the grid will clear if you set the visibility off

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-Kvmzabsk21m7bmMKN1h:b-d5cqa)*
<!-- Sent from Reviewable.io -->

dan-bowerman commented:
^ Especially for CESI's Protected Areas layer, which for performance reasons will be a tile cache, but we want to put an invisible layer "on top" for identifying/grid purposes.

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-KvmzzpPRXk2kMjeOEcd:b-uzadbm)*
<!-- Sent from Reviewable.io -->

james-rae commented:
Efficiency - When all checkboxes are checked, we are having the app evaluate all the logic for every symbol. Would be better if we can detect ALL-ON state and essentially clear the filter from the grid and layer. Especially relevant if grid is open (as we end up asking the server for all the ids, then tell the grid “hey, all these are good”).

UI - if layer is invisible, and we touch a symbol checkbox, it will turn back on the visibility. Probably should de-couple this. Was also able to get visibility controls on main legend and layer settings panel to go out-of-synch. Can also get a state where all checkboxes are turned off, but then start toggling visibility and see all points re-appear on the map.

Grid - if grid is open and all symbol filters are unchecked, the grid shows all rows. Likely caused because all-off layers results in an empty string for the definition query. The logic treats this as “no definition” and acts accordingly (show all rows).

UI - I get no checkboxes when loading a class breaks layer. Try with [Sea Parks](http://section917.cloudapp.net/arcgis/rest/services/TestData/ClassBreaks/MapServer/0).  Same with this [two-field unique value layer](http://geoappext.nrcan.gc.ca/arcgis/rest/services/NEB/Incident/MapServer/0) in feature layer mode (they appear in dynamic mode).

Efficiency - It’s looking like with feature layers it’s re-querying all geometry, which is unfortunate but is controlled by ESRI API. The network traffic looks really inefficient; I believe it’s doing queries by bounding boxes across the map, but I can see the same features coming down in multiples (so overlap). Noticeable redraw for each click.  If we deem this unacceptable, a workaround would be to essentially run the grid filter logic, and use the result array of ids to turn on/off the visibility of the actual on-client features. We no longer set the definition query on the feature layer object. Less network traffic on symbol filter change, but possibly more on every pan/zoom (would need some analysis). Needs more code to manage the feature visibility and ensure things get reapplied on pan/zoom (especially if on-demand layer, which is always re-creating on-client features).

---

Reviewed 8 of 9 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[src/app/ui/toc/templates/symbology-stack.html, line 14 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-KvnEP6OwOn3RnVyY_jC:-KvnEP6OwOn3RnVyY_jD:b7n3x90) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/7342df2a5b5c86bd18e61370efcdaa3fb4615b38/src/app/ui/toc/templates/symbology-stack.html#L14)):*
> ```HTML
>             <span class="rv-symbol-label">{{ symbol.name }}</span>
>         </div>
>         
> ```

evil spaces

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-KvnQFGIdbeMV0bTXIT4:b-neavz8)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[src/app/ui/toc/templates/symbology-stack.html, line 14 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-KvnEP6OwOn3RnVyY_jC:-Kw59uLg6m7gRercDOU8:b-rzaq68) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/7342df2a5b5c86bd18e61370efcdaa3fb4615b38/src/app/ui/toc/templates/symbology-stack.html#L14)):*
<details><summary><i>Previously, james-rae (James Rae) wrote…</i></summary><blockquote>

evil spaces
</blockquote></details>

To battle evil spaces, add this to your vscode settings:
![](https://i.imgur.com/OAPc2un.png)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418)*
<!-- Sent from Reviewable.io -->

milesap commented:
- When all symbols are checked, the query is removed instead of having a query for all symbols
- Changing symbology visibility has no effect on layer visibility
- symbology visibility is always on by default
- Fixed issue where datatable showed results when all symbols were deselected
- The Sea Parks layer now shows the symbolofy toggle boxes, however the actual toggling is a hit or a miss. It first throws a lot of errors, then after a few tries reloads and then works. 

---

Review status: 7 of 9 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-Kw5lKWK2Hnggx2HQf6r:b-ll3cbf)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
Nice, works pretty well - I like how you can toggle parts of the `Plant Hardiness` layer :+1: 

___

Errors are thrown when the details panel is opened:
> angular.js:14642 TypeError: Cannot read property 'layerRecordId' of undefined
    at Object.l [as link] (symbology-stack.directive.js:104)
    at angular.js:1385
    at angular.js:10514
    at Ae (angular.js:10520)


#### Checkboxes are badly aligned in some cases (looks like it depends on the layer's nestedness level):

###### not enough
![](https://i.imgur.com/B5zV2cJ.png)

###### too far
![](https://i.imgur.com/96zdn1b.png)

#### Sometimes checkboxes won't show up
No checbox during the opening animation here, but it will appear when closing:

![check-box.gif](https://files.reviewableusercontent.io/images/reviewable/2285779/jFsC3-Ff-hdYGFw0lQjPzupt/check-box.gif)

![check-box2.gif](https://files.reviewableusercontent.io/images/reviewable/2285779/58SXJV9I6_eXTEM-etwnjt7N/check-box2.gif)

---

Reviewed 4 of 9 files at r1.
Review status: 7 of 9 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/app/ui/toc/symbology-stack.directive.js, line 132 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-KwAJ9rK4jZxiN5sW8qL:-KwAJ9rK4jZxiN5sW8qM:btx4sar) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/5c6e3677bded78f9a61f0877e7b01dc28f44f27f/src/app/ui/toc/symbology-stack.directive.js#L132)):*
> ```JavaScript
> 
>             // trigger event which table uses to update
>             events.$broadcast('definitionClauseChgd');
> ```

Can you add this event to the events list to avoid hardcoding random strings?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-KwAJsx8VNo7Z_uljcBy:b-57pqm9)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
Reviewed 2 of 9 files at r1, 1 of 2 files at r3.
Review status: 8 of 9 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418)*
<!-- Sent from Reviewable.io -->

milesap commented:
Should be all fixed up :+1: 

---

Review status: 7 of 10 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[src/app/ui/toc/symbology-stack.directive.js, line 132 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-KwAJ9rK4jZxiN5sW8qL:-KwB-3J6IREGagfVlAPR:bb6un1j) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/5c6e3677bded78f9a61f0877e7b01dc28f44f27f/src/app/ui/toc/symbology-stack.directive.js#L132)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary><blockquote>

Can you add this event to the events list to avoid hardcoding random strings?
</blockquote></details>

Done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-KwB--KDO4HxVgFEheIS:b-at7a4f)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
When you collapse the stack, the checkbox is visible on top:
![](https://i.imgur.com/OiFXuwL.png)

When expanding, the checkboxes are rendered on top of symbols - they should be underneath them:
![](https://i.imgur.com/EZ37IVq.png)

Also, do we actually want to show a symbology checkbox if there only one item, like in here:
![](https://i.imgur.com/uNEDofg.png)

It just mimics the visiblity toggle at this point.

---

Reviewed 2 of 3 files at r4.
Review status: 9 of 10 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-KwBFpdrv8g6JWUDk2-D:b1bgdn4)*
<!-- Sent from Reviewable.io -->

milesap commented:
Removed toggles for one symbol layers and fixed that pesky lingering checkbox

---

Review status: 9 of 10 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-KwBMjLc5wtbsTy-VVjo:b-oxacsx)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
The pesky things are still there, but they are getting hidden after the transition ends. Can you move the checkboxes underneath the images, so checkboxes won't overlay the images at the start and end of the transition?

![check-box3.gif](https://files.reviewableusercontent.io/images/reviewable/2285779/FZBmNu3IZjxb_4UdsH_ialKQ/check-box3.gif)

The checkboxes are still a tiny bit off alignment, but they are getting there:
![](https://i.imgur.com/GJ1iZ7A.png)

---

Review status: 9 of 10 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-KwCGkvz7gKIlD7Yluth:b-xfoye8)*
<!-- Sent from Reviewable.io -->

james-rae commented:
Reviewed 1 of 2 files at r3, 2 of 3 files at r4, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418)*
<!-- Sent from Reviewable.io -->

milesap commented:
Alignment should be pixel perfect now. I can't make the checkbox go away before the transition ends because:
- z-index changes cause more problems
- symbol placement after the checkbox breaks layout
- **TimelineLite does not have a onReverse event??** so I can't know when a reverse animation has started. https://greensock.com/docs/TimelineLite

---

Review status: 9 of 10 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-KwGNa1ETFyfDRapz3Vy:b-1l32h3)*
<!-- Sent from Reviewable.io -->

james-rae commented:
Legend text seems to be chopping early (this is sea parks layer)
![legend.jpg](https://files.reviewableusercontent.io/images/reviewable/7585245/iU_kr4QvfFGpmuLVnlPCE34a/legend.jpeg)

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-KwGlDwhSk9QDLohfceD:b-wnqrnp)*
<!-- Sent from Reviewable.io -->

james-rae commented:
A bit of mis-match between the legend filters and the APPLY button in the datagrid.

Case 1: After adding a grid filter and clicking apply on the datagrid, then changing the legend filter, the applied filter is lost on the map.  This might be an OK behaviour, as the Apply button is sort of a one-way street. But we could also look at preserving any existing grid filter when applying legend filters. Legend guts would need a way to ask the grid if it has an active filter, and get the query string from it  (connect them with an OR i think?)

Case 2: Do some legend filters. Then do a grid filter and click apply.  The map loses the legend filters. This one is probably bad because the map doesn’t match the checkboxes in the legend. So a bit of the same, when applying the grid filter, need to tack on any legend filters that are existing.

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-KwGo4i7xymF5IrPoMpJ:b-7570fz)*
<!-- Sent from Reviewable.io -->

james-rae commented:
My bad, the legend pic above was not sea parks, it was the [incident service](http://geoappext.nrcan.gc.ca/arcgis/rest/services/NEB/Incident/MapServer/0)

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-KwGtys2qr1RmEeoBK_-:bf0cj0l)*
<!-- Sent from Reviewable.io -->

james-rae commented:
In the legend, when a grid filter is applied to the map, a small funnel icon appears on the legend entry with hovertext `Some features are hidden by filters`. Should we also apply this icon when the symbol filter is on?  Or conversely, change the hovertext to `Some features are hidden by table filters` (I like this one, since the checkboxes on the legend already give a visual queue to the symbol filter)

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-KwH3KxRsvqZBHcHvw-x:b8d8edo)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
Why do you need the `onReverse` event? Just call whatever you want after you call `.reverse();` to reverse the animation.

---

Review status: 9 of 10 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-KwZxoKzEKt3gAccol5g:badvg8g)*
<!-- Sent from Reviewable.io -->

milesap commented:
To be fair they have `onStart` so omitting `onReverse` is a head scratcher 

---

Review status: 8 of 10 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-Kw_1K7F-Ufw34pzIxgi:byw0mq6)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
Ok, if you want to show the checkboxes after the transition, can you add a small fade in/out animation to smooth things out a bit?


There are two more small problems: the "Zone" label is being chopped off and the close "X" button for the symbology stack is pushed to the left compared to the current implementation:

| new                                  | old                                  |
| ------------------------------------ | ------------------------------------ |
| ![](https://i.imgur.com/uAZyJq5.png) | ![](https://i.imgur.com/1TcLmB1.png) |

---

Review status: 8 of 10 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-Kw_2e-4nn0psFL0dYHO:be4f4al)*
<!-- Sent from Reviewable.io -->

milesap commented:
Think we're good now :crossed_fingers: 

---

Review status: 7 of 10 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-Kw_JRErwOnOIUeKr_Wg:b-o3zw78)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
Almost :D The close button is too much to the right now (maybe three or four pixels) :D we are getting there.

##### new:
![](https://i.imgur.com/5u0EwPy.png)

##### old:
![](https://i.imgur.com/2NchA2U.png)

---

Reviewed 3 of 3 files at r6.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-KwaBl_XWZE0Uy-hEIRy:b-t4rx7n)*
<!-- Sent from Reviewable.io -->

milesap commented:
I'm so close, I think I can see the light at the end of the tunnel :sun_with_face: 

---

Review status: 9 of 10 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-KwaEA3Xyt1Xhhp7DI0-:bvtk0cg)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
Nice <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

---

Reviewed 1 of 1 files at r7.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-KwaFJxOPKpsy_n0vCH5:bzhmkty)*
<!-- Sent from Reviewable.io -->

JahidAhmed commented:
Small thing I noticed, and steps to reproduce (on Sample 46 Dynamic Layer with only one child with filter -> Power Plants):
Deselect everything other than `Biomass` and `Coal` and filter the table to for example 3.
Now deselect `Coal`, all the remaining `Biomass` features appear on the map that weren't there before, but not in the table since they are filtered.

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-KwaHekpOck3JKR_FYKv:b94tl2q)*
<!-- Sent from Reviewable.io -->

JahidAhmed commented:
Clearing filters will also display all the features, even if they aren't selected. 

And also, do we want to display the filter flag in cases where we aren't showing all the symbology?

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-KwaIMJUQ7UI0XJJX6Wl:b-qncyvq)*
<!-- Sent from Reviewable.io -->

milesap commented:
Good points -  having symbol toggle and table filtering to work together would take (much?) more work but we can improve on this if the need arises (maybe for CESI). 

I'm ok with no filter flag but will add it if people complain :sun_with_face: 

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-KwaTQhJ2HHe9191056T:b-d654pn)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
Some comments up, @james-rae suggested to change the filter flag's message to `Some features are hidden by table filters` since the toggles on the symbology stack give indication that features are filtered.

Argh, I think the focus manager needs to be made aware of all these extra toggles. You kinda can tab through them but only if you start tabbing backwards after you open the stack...

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-KwjKDQGSya-2pjmTQ9R:b64fb8x)*
<!-- Sent from Reviewable.io -->

milesap commented:
Changed the filter flag text. 

Since symbology is rendered backwards in the DOM (close button is after symbols, and symbols are in reverse order) FM will now tab in reverse order starting at the bottom. On the plus side it seems like a fancy feature we spent a lot of time making. 

---

Review status: 7 of 10 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-KwjTNlm56TkKGQpaR6B:b-8c0p4x)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
When tabbing, the focus gets stack on the symbology stack and doesn't move forward. (index-samples, config 1)
![tabbbbbbing.gif](https://files.reviewableusercontent.io/images/reviewable/2285779/BBTT2SzfT3mtq2pPOT3j9bA7/tabbbbbbing.gif)

---

Review status: 7 of 10 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-KwoRRSCaAjIpFjSWv5A:buq41y2)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
re: backward rendered symbology
Here's the double-reversed version: https://github.com/AleksueiR/fgpv-vpgf/commit/03937639d8433c331007cc130a281b5cc311e06e

and the demo: http://fgpv.cloudapp.net/demo/users/AleksueiR/2368-reverse-symbology-stack-dom-order/prod/samples/index-samples.html

---

Review status: 7 of 10 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-KwoT-1daEzuLLo0_Kfd:b5abaph)*
<!-- Sent from Reviewable.io -->

james-rae commented:
Everything seems broken to me now.  I also no longer see `package.json` in the file list...did we lose our pointer to the matching geoApi?

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-KwpAVQdgnC6yets81FH:b8bvpii)*
<!-- Sent from Reviewable.io -->

milesap commented:
Fixed mysterious package.json disappearance - should be working again. 

Implemented your symbology reorder @aleksueir 

Focus manager is happy once more

---

Review status: 5 of 12 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-KwpgLAehKb41LZzQKsx:b6q8bnq)*
<!-- Sent from Reviewable.io -->

james-rae commented:
K now I need to make a change to geoApi. Turns out the where clause in a query is data-type agnostic, but in map image export it's data-type sensitive.

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-Kwq5T23rV_fo8OArHPV:b98q1ry)*
<!-- Sent from Reviewable.io -->

james-rae commented:
Reviewed 2 of 3 files at r6, 1 of 1 files at r7, 1 of 2 files at r8, 2 of 2 files at r9, 5 of 5 files at r10.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418)*
<!-- Sent from Reviewable.io -->

milesap commented:
Looks like we're good to go, final geoApi implemented  and @james-rae opened future issues to hash out:

fgpv-vpgf/fgpv-vpgf#2438
fgpv-vpgf/fgpv-vpgf#2439

---

Review status: 8 of 12 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-Kx7wCc0BLmEkESQlPK8:b5tx9fh)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>  :+1: 

good to merge?

---

Reviewed 1 of 2 files at r9, 3 of 5 files at r10, 3 of 3 files at r11, 1 of 1 files at r12.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-Kx8CGWpRreKCX3dxcbw:bxz2rem)*
<!-- Sent from Reviewable.io -->

milesap commented:
Good to go

milesap commented:
Good to go

AleksueiR commented:
I don't think this is a good idea to point to the geoapi's develop branch. This can be prone to breaking if someone will merge geoapi changes that are not yet supported by the client.

---

Reviewed 1 of 1 files at r13.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418#-:-Kx8FkI7r6H2qOjCE8o0:b8zqcnb)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
Reviewed 1 of 1 files at r14.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2418)*
<!-- Sent from Reviewable.io -->
